src/libraries/Common/src/System/Reflection/Metadata/TypeName.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace System.Reflection.Metadata
+{
+#if SYSTEM_PRIVATE_CORELIB
```

Comment:
Just to help with my ignorance, this is internal in CoreLib because you want to make it available in netstandard2.0 via `System.Reflection.Metadata` so to avoid name collision you keep the one in CoreLib private?